### PR TITLE
[GEOS-8223] Allow ordering layer contents in WMS using a "sortBy" vendor parameter, as well as mosaic contents in WCS

### PR DESCRIPTION
**Pull request by aaime**
This pull request depends on https://github.com/geotools/geotools/pull/1652 to compile, but it can be already reviewed in the meantime :-)

### CODE REVIEW

**Comment by aaime** [comment]
@dromagnoli Travis restarted and build looks good.

File: doc/en/user/source/services/wms/vendor.rst [hunk 1]
```diff
@@ -69,6 +69,27 @@ An example of a simple CQL filter is::
...
+  | In image mosaic, this causes the first granule found in the sorting will display on top, and then the others will follow.
+  | Thus, to sort a scattered mosaic of satellite images so that the most recent image shows on top, and assuming the time attribute is called ``ingestion`` in the mosaic index, the specification will be ``&sortBy=ingestion D``.
+* | For **vector layers**, ``sortBy`` maps to a sort by clause in the vector data source, and then painting happens using the normal "painter model" rules, so the first item returned is painted first, and then all others on top of it.
+  | Thus, to sort a set of event points so that the most recent event is painted on top, and assuming the attribute is called "date" in the vector layer, the specification will be ``&sortBy=date`` or ``&sortBy=date A`` (ascending direction being the default one).
+* | For **layer groups**, the sort specification is going to be copied over all internal layers, so the spec has to be valid for all of them, or an error will be reported. 
+  | An empty spec can be used for layer groups in this case, for example, ``layers=theGroup,theLayer&sortBy=(),(date A)``
+ 
+
```
Comment (jodygarnett) [inline doc/en/user/source/services/wms/vendor.rst]: thanks for the clear documentation.

File: src/wcs2_0/src/main/java/org/geoserver/wcs2_0/kvp/WCS20GetCoverageRequestReader.java [hunk 2]
```diff
@@ -1,9 +1,20 @@
 package org.geoserver.wcs2_0.kvp;
```
Comment (jodygarnett) [inline src/wcs2_0/src/main/java/org/geoserver/wcs2_0/kvp/WCS20GetCoverageRequestReader.java]: missing header - may not care if this generated code being refreshed?

Comment (aaime) [inline src/wcs2_0/src/main/java/org/geoserver/wcs2_0/kvp/WCS20GetCoverageRequestReader.java]: Header added

**Comment by aaime** [comment]
Merging, failure seems to be unrelated (cannot reproduce, did not touch anything in app-schema)

**Comment by bencaradocdavies** [comment]
@aaime `LocalResolveTest.testResolveTimeOut` uses a thread to test timeout behaviour. This type of test is vulnerable to failure on shared or virtualised platforms, and it occasionally fails. Your decision to merge was in my view correct.

**Pull request merged**